kore/src/Kore/Internal/TermLike.hs

```diff
@@ -1014,78 +1015,112 @@ forceSort forcedSort = Recursive.apo forceSortWorker
-                    and'' = and' { andSort = forcedSort }
-                OrF or' -> OrF (Right <$> or'')
-                  where
-                    or'' = or' { orSort = forcedSort }
-                IffF iff' -> IffF (Right <$> iff'')
-                  where
-                    iff'' = iff' { iffSort = forcedSort }
-                ImpliesF implies' -> ImpliesF (Right <$> implies'')
-                  where
-                    implies'' = implies' { impliesSort = forcedSort }
-                NotF not' -> NotF (Right <$> not'')
-                  where
-                    not'' = not' { notSort = forcedSort }
-                NextF next' -> NextF (Right <$> next'')
-                  where
-                    next'' = next' { nextSort = forcedSort }
-                RewritesF rewrites' -> RewritesF (Right <$> rewrites'')
-                  where
-                    rewrites'' = rewrites' { rewritesSort = forcedSort }
-                ExistsF exists' -> ExistsF (Right <$> exists'')
-                  where
-                    exists'' = exists' { existsSort = forcedSort }
-                ForallF forall' -> ForallF (Right <$> forall'')
-                  where
-                    forall'' = forall' { forallSort = forcedSort }
-                -- Rigid: These patterns should never have sort _PREDICATE{}.
-                MuF _ -> illSorted
-                NuF _ -> illSorted
-                ApplySymbolF _ -> illSorted
-                ApplyAliasF _ -> illSorted
-                BuiltinF _ -> illSorted
-                DomainValueF _ -> illSorted
-                StringLiteralF _ -> illSorted
-                VariableF _ -> illSorted
-                InhabitantF _ -> illSorted
+
+{-| Attempts to modify the pattern to have the given sort, ignoring the
+previous sort and without assuming that the pattern's sorts are consistent.
+-}
+fullyOverrideSort
+    :: forall variable
+    .  (SortedVariable variable, Unparse variable, GHC.HasCallStack)
+    => Sort
+    -> TermLike variable
+    -> TermLike variable
+fullyOverrideSort forcedSort = Recursive.apo overrideSortWorker
+  where
+    overrideSortWorker
+        :: TermLike variable
+        -> Base
+            (TermLike variable)
+            (Either (TermLike variable) (TermLike variable))
+    overrideSortWorker original@(Recursive.project -> attrs :< _) =
+        (:<)
+            (attrs { Attribute.patternSort = forcedSort })
+            (forceSortWorkerPredicate forcedSort original)
+
+illSorted
+    :: (SortedVariable variable, Unparse variable, GHC.HasCallStack)
+    => Sort -> TermLike variable -> a
+illSorted forcedSort original =
+    (error . show . Pretty.vsep)
+    [ Pretty.cat
+        [ "Could not force pattern to sort "
+        , Pretty.squotes (unparse forcedSort)
+        , ", instead it has sort "
+        , Pretty.squotes (unparse (termLikeSort original))
+        , ":"
+        ]
+    , Pretty.indent 4 (unparse original)
+    ]
+
+forceSortWorkerPredicate
```

Comment:
I was wondering whether this is a good name.

Reply:
Better now?